### DISCUSSION TITLE
SVG has more colors than original image

### DISCUSSION THREAD

**Discussion by christianhanne** [visioncortex/vtracer#101] [Q&A]
Hello there, really love vtracer so far. Unfortunately I ran into a problem with the color precision.

I have a png image with 16 unique colors. When I use the webapp to convert it the resulting image has 1208 colors. Is there any way to tell vtracer to keep the original colors? Adjusting color precision does not seem to work.

Any help would be greatly appreciated.

Cheers

**Comment by tyt2y3** [comment]
Hi there,

I think it’s the third time I am asked about this.

I wonder what’s the use case for it?

There’s been a design we discussed allowing users to supply a color palette separately to constraint the colors to be used in the output image.

Would you like to sponsor the development of this feature?
________________________________
From: Christian Hanne ***@***.***>
Sent: Sunday, December 15, 2024 8:58:42 AM
To: visioncortex/vtracer ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [visioncortex/vtracer] SVG has more colors than original image (Discussion #101)


Hello there, really love vtracer so far. Unfortunately I ran into a problem with the color precision.

I have a png image with 16 unique colors. When I use the webapp to convert it the resulting image has 1208 colors. Is there any way to tell vtracer to keep the original colors? Adjusting color precision does not seem to work.

Any help would be greatly appreciated.

Cheers

**Comment by christianhanne** [comment]
My use case is that I have an editor where users can redefine colors in an svg file to customize logos for example. The images normally have 4 to 12 different colors. I often receive the original files as raster images and want to automate parts of this workflow in the app. Currently I'm converting them to vector by hand and then upload them into the app which is very time consuming. There are apis that are actually pretty decent, but I was hoping to find something that I could host myself and basically give control over the process to the user. Sponsoring this would probably not be viable though.

**Comment by illfygli** [comment]
Another use-case here: I'm creating SVGs for my website, and would like them to match the website theme.

I first run the source images (processed photos) through `magick`, and `-remap` the colors onto the theme palette, which gives another bitmap that only uses the theme colors. Then I run `vtracer` on that bitmap.

I'm trying to find my way around the source code to see if I can help.
